gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/LoggingServiceV2Api.java

```diff
@@ -0,0 +1,583 @@
+   * a log_name resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String createLogNamePath(String project, String log) {
+    return LOG_NAME_PATH_TEMPLATE.instantiate(
+        "project", project,"log", log);
+  }
+
+
+  /**
+   * Extracts the project from the given fully-qualified path which
+   * represents a logName resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String extractProjectFromLogNamePath(String logNamePath) {
+    return LOG_NAME_PATH_TEMPLATE.parse(logNamePath).get("project");
+  }
+
+  /**
+   * Extracts the log from the given fully-qualified path which
+   * represents a logName resource.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public static final String extractLogFromLogNamePath(String logNamePath) {
+    return LOG_NAME_PATH_TEMPLATE.parse(logNamePath).get("log");
+  }
+
+
+  // =============
+  // Service Calls
+  // =============
+
+  // ----- deleteLog -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes a log and all its log entries.
+   * The log will reappear if it receives new entries.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param logName Required. The resource name of the log to delete.  Example:
+   * `"projects/my-project/logs/syslog"`.
+   */
+  public void deleteLog(String logName) {
+    DeleteLogRequest request =
+        DeleteLogRequest.newBuilder()
+        .setLogName(logName)
+        .build();
+
+    deleteLog(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes a log and all its log entries.
+   * The log will reappear if it receives new entries.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   */
+  public void deleteLog(DeleteLogRequest request) {
+    deleteLogCallable().call(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes a log and all its log entries.
+   * The log will reappear if it receives new entries.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public ApiCallable<DeleteLogRequest, Empty> deleteLogCallable() {
+    return ApiUtils.prepareIdempotentCallable(DELETE_LOG, settings).bind(channel);
+  }
+
+  // ----- writeLogEntries -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Writes log entries to Cloud Logging.
+   * All log entries in Cloud Logging are written by this method.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param logName Optional. A default log resource name for those log entries in `entries`
```

Comment:
what happens when no logName is provided and some entries do not specify their own `logName`.